### PR TITLE
Refactor and cleanup

### PR DESCRIPTION
**Pull request by geek**
- Customers no longer depends on express
- Moved everything to strict mode and fixed global variable leaks
- Refactored code to use ES6 language features
- Updated to alpine 3.4
- Updated containerpilot to the latest version
- Customers is using [`piloted`](https://www.npmjs.com/package/piloted) module
- Sales is using [`consulite`](https://www.npmjs.com/package/consulite) module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,26 +4,55 @@ Applications on Autopilot
 *A demonstration of the autopilot pattern for self-operating microservices, designed as a workshop or tutorial.*
 
 
-Microservices and Docker go together like chocolate and peanut butter: microservice architectures provide organizations a tool to manage complexity of the development process, and application containers provide a new means to manage the dependencies and deployment of those microservices. But deploying and connecting those services together is still a challenge because it forces developers to design for operationalization.
+Microservices and [Docker](https://www.docker.com) go together like chocolate
```
Comment (tgross) [inline README.md]: On our autopilotpattern repos we don't wrap the markdown at 80 cols. Call it a Product-team quirk.

**Comment by tgross** [comment]
Minor nit over the README formatting change but this LGTM.

File: sales/sales.js [hunk 2]
```diff
@@ -1,99 +1,83 @@
...
-var salesData = require('./data');
-var consul = require('./consul');
+const Consulite = require('consulite');
+const Express = require('express');
+const Customers = require('./lib/customers');
+const Data = require('./lib/data');
 
-var app = express();
-var customersHosts = [];
+const app = Express();
```
Comment (tgross) [inline sales/sales.js:8]: Why remove the `Express` dependency from Customers but leave it in for Sales?

File: docker-compose.yml [hunk 3]
```diff
@@ -17,6 +17,9 @@ sales:
     image: autopilotpattern/workshop-sales
     links:
       - consul
+    environment:
+      - CONSUL_HOST=consul
```
Comment (tgross) [inline docker-compose.yml:21]: With Triton CNS we can get rid of the `links` field and replace this line with an `_env` file that has something like `CONSUL_HOST=consul.svc.${TRITON_ACCOUNT}.${TRITON_DC}.cns.joyent.com`. Given that Consulite has a safe default value maybe we should go:

``` yaml
...
env_file: _env
environment:
  - CONSUL_HOST
  - CONSUL_PORT=8500
...
```

This way if the user's environment has a `CONSUL_HOST` field it'll automatically be passed-thru. (We'll want to leave the `links` in for the `local-compose.yml` file though.)

File: customers/package.json [hunk 4]
```diff
@@ -3,12 +3,10 @@
...
   "main": "customers.js",
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
-  },
+  "scripts": {},
   "author": "Tim Gross, Joyent",
   "license": "Mozilla Public License 2.0",
   "dependencies": {
-    "express": "^4.13.4"
+    "consulite": "1.x.x"
```
Comment (tgross) [inline customers/package.json]: Should Consulite be moved under the Joyent org @misterbisson?

File: docker-compose.yml (hunk 3 above)
Comment (geek) [reply docker-compose.yml:21]: Good idea. The next PR will be to more closely integrate with triton CNS and remove the links.

File: sales/sales.js (hunk 2 above)
Comment (geek) [reply sales/sales.js:8]: This is to demonstrate that the framework doesn't matter. I also want to add another service that uses hapi + mongo.

Comment (tgross) [reply sales/sales.js:8]: Hrm, one of the big advantages of this blueprint relative to the others is that the whole thing is pretty small and easy to understand. Adding more services and dependencies and complexity doesn't jive with that.

Comment (geek) [reply sales/sales.js:8]: Makes sense. We do want to demonstrate that mongodb works well with ContainerPilot... we were thinking of adding a service in the workshop, but maybe we should do a separate standalone service.

File: customers/package.json (hunk 4 above)
Comment (geek) [reply customers/package.json]: @tgross moved to https://github.com/joyent/consulite

**Pull request merged**